src/arcade/potts/env/location/PottsLocation.java

```diff
@@ -659,6 +661,19 @@ Location separateVoxels(
         return makeLocation(voxelsB);
     }
 
+    /**
+     * Swaps the voxels in this location with the voxels in another location.
+     *
+     * @param location the location to swap with
+     */
+    public void swapVoxels(PottsLocation location) {
```

Comment:
I think this state change will be somewhat difficult to track and prone to bugs - an object changes both itself and another object, implicitly.  is this necessary? maybe a static method makes more sense here?

Reply:
I try to avoid side effects with static methods but that might just be arbitrary. Fine with whatever you decide here.